### ISSUE TITLE
EsmTokenizer does not return correct length token type ids

### ISSUE THREAD

**Issue by lhallee** [huggingface/transformers#28656]
### System Info

transformers 4.37
python 3.10.11

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

EsmTokenizer ignores special tokens when generating

```
tokenizer = EsmTokenizer.from_pretrained('facebook/esm2_t6_8M_UR50D')
seq = 'PROTEIN'
len(seq) # 7

tokens = tokenizer(seq, seq, return_token_type_ids=True)

len(tokens.input_ids), len(tokens.token_type_ids) # 17, 14 
```

Is the same for return_special_tokens=True or False

I understand Esm does not natively use token_type_ids but some personal versions and upcoming contributions to the field do and could benefit from having EsmTokenizer with correct token_type_ids.

### Expected behavior

```
tokenizer = EsmTokenizer.from_pretrained('facebook/esm2_t6_8M_UR50D')
seq = 'PROTEIN'
len(seq) # 7

tokens = tokenizer(seq, seq, return_token_type_ids=True)

len(tokens.input_ids), len(tokens.token_type_ids) # 17, 17 
```

**Comment by lhallee** [comment]
Suggested changes that fix this in Pull request [#28657](https://github.com/huggingface/transformers/pull/28657)

**Comment by amyeroberts** [comment]
cc @Rocketknight1
